lib/galaxy/managers/jobs.py

```diff
@@ -220,9 +213,14 @@ def add_search_criteria(stmt):
             if user_details:
                 stmt = stmt.outerjoin(Job.user)
         else:
-            if history_id is None and invocation_id is None and implicit_collection_jobs_id is None and trans.user:
-                stmt = stmt.where(Job.user_id == trans.user.id)
-            # caller better check security
+            if history_id is None and invocation_id is None and implicit_collection_jobs_id is None:
+                # If we're not filtering on history, invocation or collection we filter the jobs owned by the current user
+                if trans.user:
+                    stmt = stmt.where(Job.user_id == trans.user.id)
+                elif trans.galaxy_session:
+                    stmt = stmt.where(Job.session_id == trans.galaxy_session.id)
+                else:
+                    return None
```

Comment:
Thanks, nice improvement! Going back to my original point, maybe raise an exception here?
It looks to me that an API client should either pass one of the 3 IDs or have at least an active session.

Reply:
Alright, I can do that, no-op seems fine, but I guess an error message will be more informative for API users in case they miss the parameter accidentally.